AdvanceTaskMarsPart1/Tests/Language_Tests.cs

```diff
@@ -0,0 +1,180 @@
+﻿using AdvanceTaskMarsPart1.Data;
+using AdvanceTaskMarsPart1.Pages;
+using AdvanceTaskMarsPart1.Utilities;
+using AventStack.ExtentReports;
+using AventStack.ExtentReports.Reporter;
+using NUnit.Framework;
+
+namespace AdvanceTaskMarsPart1.Tests
+{
+    public class Language_Tests : CommonDriver
+    {
+        LoginPage loginPageObj;
+        LanguagePage languagePageObj;
+        public static ExtentReports extent;
+        public static ExtentTest test;
+
+        public Language_Tests()
+        {
+            loginPageObj = new LoginPage();
+            languagePageObj = new LanguagePage();
+        }
+
+        [OneTimeSetUp]
+        public static void ExtentStart()
+        {
+            //Create new instance of ExtentReports
+            extent = new ExtentReports();
+            //Create new instance of ExtentSparkReporter
+            var SparkReporter = new ExtentSparkReporter(@"D:\Sasikala\MVP_Studio\AdvanceTaskPart1\AdvanceTaskMarsPart1\AdvanceTaskMarsPart1\ExtentReports\Language.html");
+            //Attach the ExtentSparkReporter to the ExtentReports
+            extent.AttachReporter(SparkReporter);
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            Initialize();
+            loginPageObj.LoginActions();
+        }
+
+        [Test, Order(1), Description("This test is deleting all records in language list")]
+        public void Delete_All_Records()
+        {
+            test = extent.CreateTest("Delete_AllRecords").Info("Test started");
+            languagePageObj.Delete_All_Records();
+            test.Log(Status.Pass, "Delete_AllRecords passed");
+        }
+
+        [Test, Order(2), Description("This test is adding language in the language list")]
+        public void Add_Language()
+        {
+            test = extent.CreateTest("Add_Language").Info("Test started");
+            // Read test data for the AddLanguage test case
+            List<LanguageData> languageDataList = LanguageDataHelper.ReadLanguageData(@"addLanguageData.json");
+
+            // Iterate through test data and retrieve AddLanguage test data
+            foreach (var languageData in languageDataList)
+            {
+                languagePageObj.Add_Language(languageData);
+                string actualMessage = languagePageObj.getMessage();
+
+                // Check if the Language contains special characters
+                bool containsSpecialCharacters = ContainsSpecialCharacters(languageData.Language);
+                if(containsSpecialCharacters)
+                {
+                    try
+                    {
+                        // Verify that the actual message matches the expected message for special characters
+                        Assert.That(actualMessage == languageData.ExpectedMessage, "Actual message and expected message do not match");
```

Comment:
Asserts and steps should be kept moved into packages! I think before I review further you should update the framework to follow the example i have shown in the task spec

Reply:
Asserts**